src/forms/dynamicXone.js

```diff
@@ -0,0 +1,79 @@
+import { decorate } from '@hydrofoil/shaperone-wc/templates.js'
+import { findNodes } from 'clownface-shacl-path'
+import { isGraphPointer } from 'is-graph-pointer'
+import { owl } from '@tpluscode/rdf-ns-builders'
+import sh1 from '@hydrofoil/shaperone-core/ns.js'
+
+/**
+ * Focus node rendering decorator, which will choose properties to render by only showing those
+ * matching a `sh1:xoneDiscriminator`.
+ *
+ * Add a hidden property to the items of `sh:xone`. Properties within its group will be shown onyl when
+ * the focus node value of the discriminator matches the `sh:hasValue` of that hidden property
+ *
+ * <shape>
+ *    sh1:discriminator <property> ;
+ *    sh:xone (
+ *      [
+ *        sh:property [
+ *          sh:path <property> ;
+ *          sh:hasValue "foo" ;
+ *          dash:hidden true ;
+ *        ] ;
+ *        sh:property <foo1> , <foo2> , <foo3>;
+ *      ]
+ *      [
+ *        sh:property [
+ *          sh:path <property> ;
+ *          sh:hasValue "bar" ;
+ *          dash:hidden true ;
+ *        ] ;
+ *        sh:property <bar1> , <bar2> , <bar3>;
+ *      ]
+ *    )
+ * .
+ */
```

Comment:
I'm still confused by the :xoneDiscriminator. 

But not only the name. I fail to see what benefits it brings to have this declaration to the turtle.

Reply:
Wouldn't it be more straightforward to have some code to do custom visuals? 

One implementation could be with the code vocab.

```
<someProp> sh1:hasCustom <sh1:debugProperty> .

<sh1:debug>
    a                  sh1:CustomRule ;
    code:implementedBy [ a         code:EcmaScriptModule ;
                         code:link <file:./lib/shouldDisplay.js> ] ;
    code:arguments     ....more custom stuff.


function shouldDisplay({ pointer, vars}){
	// do some magic
	return true
} 
```

Reply:
I'm saying it because adding new semantics to the sh1: namespace will it more complex in the longer, which means more maintenance.

Reply:
For one, this is not at all using `code:` vocabulary and loaders, which are node-only for now.

That notwithstanding, you eventually bundle the app with webpack, or whatever webcomponents.dev uses in the background for example, you cannot easily set up a scenario in which code is directly referred from turtle.

> I fail to see what benefits it brings to have this declaration in the turtle.

Which one? The `xoneDiscriminator`? I find it more clear when it's declarative. The `sh1:xoneDiscriminator` is clearly rooted in the data in how it control the logic of hiding or displaying based on the focus node's properties. By keeping it turtle you make it descriptive instead of prescriptive. 

Happy to discuss other ideas. The extensibility model of shaperone is not yet what it needs to be overall (cue the awkward `sh1:showInDebugOnly` predicate which indeed should not exist 😄)